pkg/ansible/watches/watches.go

```diff
@@ -121,9 +135,55 @@ func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
-// Load - loads a slice of Watches from the watch file at `path`.
-func Load(path string) ([]Watch, error) {
-	b, err := ioutil.ReadFile(path)
+// Validate - ensures that a Watch is valid
+// A Watch is considered valid if it:
+// - Specifies a valid path to a Role||Playbook
+// - If a Finalizer is non-nil, it must have a name + valid path to a Role||Playbook or Vars
+func (w *Watch) Validate() error {
+	err := verifyAnsiblePath(w.Playbook, w.Role)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Invalid ansible path for GVK: %v", w.GroupVersionKind.String()))
+		return err
+	}
+
+	if w.Finalizer != nil {
+		if w.Finalizer.Name == "" {
+			err = fmt.Errorf("finalizer must have name")
+			log.Error(err, fmt.Sprintf("Invalid finalizer for GVK: %v", w.GroupVersionKind.String()))
+			return err
+		}
+		// only fail if Vars not set
+		err = verifyAnsiblePath(w.Finalizer.Playbook, w.Finalizer.Role)
+		if err != nil && len(w.Finalizer.Vars) == 0 {
+			log.Error(err, fmt.Sprintf("Invalid ansible path on Finalizer for GVK: %v", w.GroupVersionKind.String()))
+			return err
+		}
+	}
+
+	return nil
+}
+
+// New - returns a Watch with sensible defaults.
+// The returned Watch is not valid as it does not include a role or playbook.
+func New(gvk schema.GroupVersionKind) *Watch {
```

Comment:
-1 I don't think we should have a `New` function that does not return a valid watch. I also didn't see where this was used outside of a test.
```bash
[jesusr@camaro ansible{foo}]$ git grep watches.New
runner/runner_test.go:                  testWatch := watches.New(tc.gvk)

[jesusr@camaro ansible{foo}]$ cd watches/
[jesusr@camaro watches{foo}]$ git grep "New("
watches.go:func New(gvk schema.GroupVersionKind) *Watch {
watches.go:             return errors.New("version must not be empty")
watches.go:             return errors.New("kind must not be empty")
watches_test.go:func TestNew(t *testing.T) {
watches_test.go:        watch := New(expectedGVK)
```

If you want a convenience method to create a struct in a test, I'd add it to the test. When I see a `New` function I think of it as a constructor and expect to get a valid struct.

Reply:
Since I made all of the default values private to the package. If a go-based operator developer wants to create a runner for his playbook, he/she will need a way to get a `watch` with sensible defaults.

Assuming the `New()` stays, I understand this comment to mean that it should accept all the arguments required for a valid `watch` to be returned. Is that right?

Reply:
Made it so it's possible to create a valid `watch`.